src/dask_awkward/manual/column_optimization.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+from typing import cast
+
+from dask.highlevelgraph import HighLevelGraph
+
+from dask_awkward.layers.layers import AwkwardInputLayer
+from dask_awkward.lib.core import Array
+
+
+def optimize_columns(array: Array, columns: dict[str, frozenset[str]]) -> Array:
```

Comment:
We need to think of way to ensure that the resulting output is not optimised again at compute time.

Reply:
That's why I set `.is_projectable` on the layers that are optimized "by hand" to `False`. However, I'm not sure if that's the most elegant solution.

Reply:
Hah, OK, I can see that works. We should see what it does with multiple input layers.

Reply:
:+1: It should work as follows when calling `.compute(optimize_graph=True)`: any layer that has been manually projected won't be optimized again (they are basically treated as a any other non-column-optimizable layer), whereas any other AwkwardInputLayer goes through the usual automatic optimization process.